### DISCUSSION TITLE
Possibility to use PostSharp instead of Fody?

### DISCUSSION THREAD

**Discussion by CyberBotX** [reactiveui/ReactiveUI#3097] [Q&A] [answered]
Hi, I have been looking into migrating to using ReactiveUI for some of my WPF projects, and I noticed that ReactiveUI's solution to reducing boilerplate code was to use Fody. I personally use PostSharp in a lot of my projects and I was wondering if anyone has PostSharp equivalents to the Fody solution that ReactiveUI uses.

I have personally only been able to convert the `Reactive` attribute, albeit I had to manually perform the code of `RaiseAndSetIfChanged` inside of my PostSharp aspect's code, as while PostSharp can access the compiler generated backing variable, it doesn't appear to have a way of doing so that allows it to be passed to `RaiseAndSetIfChanged` with the `ref` keyword.

I haven't been able to find a way to replace the Fody version of the `ObervableAsProperty` attribute with a PostSharp one, mostly because I do not know if PostSharp has a way to replace the compiler generated backing variable with a different one, which would make it practically impossible to implement a version of the `ToPropertyEx` method that can access the PostSharp version of the backing field, and even if one could somehow be added alongside the existing one (also something I am unsure about), it doesn't appear that it can be done in a way that `ToPropertyEx` can access it.

This is mostly for curiosity sake, as I have found that I can include both Fody and PostSharp in the same application without any issues, but I also suspect that my reliance on PostSharp with a ReactiveUI -based project will be a lot less than with a project not using ReactiveUI.

**Comment by glennawatson** [comment, marked as answer]
I don't think introducing another IL modifier framework is in the future since there is already a bit of a maintenance burden for us with the fody one.
